sequencer/src/transaction.rs

```diff
@@ -21,21 +20,14 @@ use serde::{de::Error, Deserialize, Deserializer, Serialize};
 ///   module because that's where it's byte length is dictated, so that's where
 ///   it makes the most sense to put serialization. See
 ///   <https://github.com/EspressoSystems/espresso-sequencer/pull/1499#issuecomment-2134065090>
-/// - It impls [`Namespace`] from [`jf_merkle_tree`], but this seems unneeded
-///   now that we're not using jellyfish's namespace merkle tree.
-/// - We derive lots of things that perhaps we shouldn't: `Into`, `From`,
-///   `Default`, `Ord`. Perhaps derivations for [`NamespaceId`] should match
-///   that of [`Transaction`].
 #[derive(
     Clone,
-    Copy,
```

Comment:
Why shouldn't this be copyable?

Reply:
`Copy` restored in https://github.com/EspressoSystems/espresso-sequencer/pull/1607/commits/b01f5717cd8f1fde6805a3c2131666d7a619212c.

This is a matter of philosophy. I like my types to promise as little as possible. `Copy` suggests that cloning this type will always be a fast bitwise copy. That's probably true, but what's the benefit of committing to that now?

For a different perspective, [official Rust doc](https://doc.rust-lang.org/std/marker/trait.Copy.html#when-should-my-type-be-copy) says "Generally speaking, if your type *can* implement `Copy`, it should" and then immediately proceeds to a warning about making promises you can't keep. 🤷‍♂️